lib/watson-discovery-setup.js

```diff
@@ -0,0 +1,309 @@
+  return new Promise((resolve, reject) => {
+    this.discoveryClient.getCollections(params, (err, data) => {
+      if (err) {
+        console.error(err);
+        return reject(new Error('Failed to get Discovery collections.'));
+      } else {
+        // If a DISCOVERY_COLLECTION_ID is set, validate it and use it (or fail).
+        // Otherwise, look (by name) for one that we already created.
+        const validateID = process.env.DISCOVERY_COLLECTION_ID;
+        const DISCOVERY_COLLECTION_NAME = process.env.DISCOVERY_COLLECTION_NAME || params.default_name;
+        const collections = data.collections;
+        for (let i = 0, size = collections.length; i < size; i++) {
+          const collection = collections[i];
+          if (validateID) {
+            if (validateID === collection.collection_id) {
+              console.log('Found Discovery collection using DISCOVERY_COLLECTION_ID.');
+              console.log(collection);
+              params.collection_name = collection.name;
+              params.collection_id = collection.collection_id;
+              return resolve(params);
+            }
+          } else if (collection.name === DISCOVERY_COLLECTION_NAME) {
+            console.log('Found Discovery collection by name.');
+            console.log(collection);
+            params.collection_name = collection.name;
+            params.collection_id = collection.collection_id;
+            return resolve(params);
+          }
+        }
+        if (validateID) {
+          return reject(new Error('Configured DISCOVERY_COLLECTION_ID=' + validateID + ' not found.'));
+        }
+        // No collection_id added, but return params dict. Set the name to use to create a collection.
+        params.collection_name = DISCOVERY_COLLECTION_NAME;
+        return resolve(params);
+      }
+    });
+  });
+};
+
+/** Create a Discovery environment if we did not find one.
+ * If an environment is passed in, then we already have one.
+ * When we create one, we have to create it with our known name
+ * so that we can find it later.
+ * @param {Object} params - Object describing the environment we found or need.
+ * @return {Promise} Promise with resolve(environment) or reject(err).
+ */
+WatsonDiscoverySetup.prototype.createDiscoveryEnvironment = function(params) {
+  if (params.environment_id) {
+    return Promise.resolve(params); // If we have an ID, then the env must exist.
+  }
+  return new Promise((resolve, reject) => {
+    // No existing environment found, so create it.
+    // NOTE: The number of environments that can be created
+    // under a trial Bluemix account is limited to one per
+    // organization. That is why have the "reuse" strategy above.
+    console.log('Creating discovery environment...');
+    const createParams = {
+      name: params.environment_name,
+      description: 'Discovery environment created by ' + params.default_name,
+      size: '0' // Use string to avoid default of 1.
```

Comment:
FYI -- I believe the SDK was fixed to allow 0 instead of '0'. Either way be sure to have the SDK pinned.